src/convertToObject.js

```diff
@@ -3,16 +3,31 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles (see an example in [stylesString.js]
+ * (./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties (see an exampl in
+ * [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const sourceObject = sourceString.trim().split(';')
+    .map(property => property.trim().split(':'))
+    .filter(property => property[0] !== '')
+    .reduce((properties, property) => {
+      const keyProperty = property[0].trim();
+      const valueProperty = property[1].trim();
+
+      return {
+        ...properties,
+        [keyProperty]: valueProperty,
+      };
```

Comment:
```suggestion
    .reduce((properties, [keyProperty,  valueProperty]) => {
      return {
        ...properties,
        [keyProperty.trim()]: valueProperty.trim(),
      };
```
Also, you can use destructuring, something like this